pilot/pkg/config/kube/crdclient/client.go

```diff
@@ -230,6 +232,28 @@ func (cl *Client) Update(config config.Config) (string, error) {
 	return meta.GetResourceVersion(), nil
 }
 
+// PatchFunc provides the cached config as a base for modification. The diff between the input value and the modified
+// return will be used to apply a patch in Kubrenetes.
+type PatchFunc func(cfg config.Config) config.Config
+
+// Patch applies only the modifications made in the PatchFunc rather than doing a full replace. Useful to avoid
+// read-modify-write conflicts when there are many concurrent-writers to the same resource.
+func (cl *Client) Patch(typ config.GroupVersionKind, name, namespace string, patchFn PatchFunc) (string, error) {
+	orig := cl.Get(typ, name, namespace)
```

Comment:
We can deal with pretty much any amount of stale-ness on the `Get`. The diff is what we actually send for the update.